src/ray/raylet/node_manager.cc

```diff
@@ -2565,6 +2639,55 @@ void NodeManager::RecordMetrics() {
   object_directory_->RecordMetrics(duration_ms);
 }
 
+void NodeManager::ConsumeSyncMessage(
+    std::shared_ptr<const syncer::RaySyncMessage> message) {
+  if (message->component_id() == syncer::RayComponentId::RESOURCE_MANAGER) {
+    rpc::ResourcesData data;
+    data.ParseFromString(message->sync_message());
+    NodeID node_id = NodeID::FromBinary(data.node_id());
+    UpdateResourceUsage(node_id, data);
+  } else if (message->component_id() == syncer::RayComponentId::COMMANDS) {
+    rpc::ResourcesData data;
+    data.ParseFromString(message->sync_message());
+    if (data.should_global_gc()) {
+      should_local_gc_ = true;
+    }
+  }
+}
+
+std::optional<syncer::RaySyncMessage> NodeManager::CreateSyncMessage(
```

Comment:
will syncer::RayComponentId ever change for the same component? if it doesn't change, should we remove it from the interface?

similarly, does after_version ever change? maybe we can make it default to 0.

Reply:
I don't think we can remove this otherwise, it will couple with RaySyncer.

For 1) if we don't give RayComponentId, then we can't have one class serve as two components here. For example, for node manager, it needs to send messages for GC and also loads for now. Ideally, it's not necessary because we should have different classes for different purposes. So I think it can be removed in the future once we get a better design. 

For 2) I don't think so. Otherwise, you have to store the last checkpoint data inside the class which introduce a state. I feel store the state in RaySyncer is much cleaner.